src/renderers/WebGLRenderer.js

```diff
@@ -47,21 +47,30 @@ function createCanvasElement() {
 
 }
 
-function WebGLRenderer( parameters ) {
-
-	parameters = parameters || {};
-
-	const _canvas = parameters.canvas !== undefined ? parameters.canvas : createCanvasElement(),
-		_context = parameters.context !== undefined ? parameters.context : null,
-
-		_alpha = parameters.alpha !== undefined ? parameters.alpha : false,
-		_depth = parameters.depth !== undefined ? parameters.depth : true,
-		_stencil = parameters.stencil !== undefined ? parameters.stencil : true,
-		_antialias = parameters.antialias !== undefined ? parameters.antialias : false,
-		_premultipliedAlpha = parameters.premultipliedAlpha !== undefined ? parameters.premultipliedAlpha : true,
-		_preserveDrawingBuffer = parameters.preserveDrawingBuffer !== undefined ? parameters.preserveDrawingBuffer : false,
-		_powerPreference = parameters.powerPreference !== undefined ? parameters.powerPreference : 'default',
-		_failIfMajorPerformanceCaveat = parameters.failIfMajorPerformanceCaveat !== undefined ? parameters.failIfMajorPerformanceCaveat : false;
+function WebGLRenderer( {
+	canvas = createCanvasElement(),
+	context = null,
+	alpha = false,
+	depth = true,
+	stencil = true,
+	antialias = false,
+	premultipliedAlpha = true,
+	preserveDrawingBuffer = false,
+	powerPreference = 'default',
+	failIfMajorPerformanceCaveat = false
+} ) {
+
+	const _canvas = canvas,
```

Comment:
I don't understand how how this works. Where is `canvas` defined?

Reply:
I run `npm start` and saw `build/three.js`, it looks like this.
It looks odd, I think I should close this PR.

```javascript
	function WebGLRenderer(_ref) {
		var _ref$canvas = _ref.canvas,
				canvas = _ref$canvas === void 0 ? createCanvasElement() : _ref$canvas,
				_ref$context = _ref.context,
				context = _ref$context === void 0 ? null : _ref$context,
				_ref$alpha = _ref.alpha,
				alpha = _ref$alpha === void 0 ? false : _ref$alpha,
				_ref$depth = _ref.depth,
				depth = _ref$depth === void 0 ? true : _ref$depth,
				_ref$stencil = _ref.stencil,
				stencil = _ref$stencil === void 0 ? true : _ref$stencil,
				_ref$antialias = _ref.antialias,
				antialias = _ref$antialias === void 0 ? false : _ref$antialias,
				_ref$premultipliedAlp = _ref.premultipliedAlpha,
				premultipliedAlpha = _ref$premultipliedAlp === void 0 ? true : _ref$premultipliedAlp,
				_ref$preserveDrawingB = _ref.preserveDrawingBuffer,
				preserveDrawingBuffer = _ref$preserveDrawingB === void 0 ? false : _ref$preserveDrawingB,
				_ref$powerPreference = _ref.powerPreference,
				powerPreference = _ref$powerPreference === void 0 ? 'default' : _ref$powerPreference,
				_ref$failIfMajorPerfo = _ref.failIfMajorPerformanceCaveat,
				failIfMajorPerformanceCaveat = _ref$failIfMajorPerfo === void 0 ? false : _ref$failIfMajorPerfo;
		var _canvas = canvas,
				_context = context,
				_alpha = alpha,
				_depth = depth,
				_stencil = stencil,
				_antialias = antialias,
				_premultipliedAlpha = premultipliedAlpha,
				_preserveDrawingBuffer = preserveDrawingBuffer,
				_powerPreference = powerPreference,
				_failIfMajorPerformanceCaveat = failIfMajorPerformanceCaveat;
```